### ISSUE TITLE
oci-provider misses id field in resources.json for buckets

### ISSUE THREAD

**Issue by Hotpirsch** [cloud-custodian/cloud-custodian#8756]
### Describe the bug

When I wanted to create a report over oci.bucket resources the report generation fails with this error:
Traceback (most recent call last):
  File "/usr/local/bin/custodian", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/lib/python3.11/site-packages/c7n/cli.py", line 363, in main
    command(config)
  File "/usr/local/lib/python3.11/site-packages/c7n/commands.py", line 128, in _load_policies
    return f(options, list(policies))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/c7n/commands.py", line 337, in report
    do_report(
  File "/usr/local/lib/python3.11/site-packages/c7n/reports/csvout.py", line 98, in report
    rows = formatter.to_csv(records, unique=not options.all_findings)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/c7n/reports/csvout.py", line 220, in to_csv
    uniq = self.uniq_by_id(records)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/c7n/reports/csvout.py", line 202, in uniq_by_id
    rec_id = rec[self._id_field]
             ~~~^^^^^^^^^^^^^^^^
KeyError: 'id'

Looking at the `resources.json` file I noticed that none of the json constructs for a bucket had a field named "id" (unless compared to e. g. oci.user).

### What did you expect to happen?

I expected a report to be generated in the same manner as for the oci.user which I had done before without problems.

### Cloud Provider

Oracle Cloud (OCI)

### Cloud Custodian version and dependency information

```shell
Custodian:   0.9.28
Python:      3.11.4 (main, Jun 20 2023, 16:59:59) [Clang 14.0.3 (clang-1403.0.22.14.1)]
Platform:    posix.uname_result(sysname='Darwin', nodename='XXXXXXXXXX', release='22.5.0', version='Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:22 PDT 2023; root:xnu-8796.121.3~7/RELEASE_X86_64', machine='x86_64')
Using venv:  False
Docker: False
Installed: 

PyJWT==2.7.0
adal==1.2.7
applicationinsights==0.11.10
apscheduler==3.10.1
argcomplete==3.1.1
attrs==23.1.0
azure-common==1.1.28
azure-core==1.27.1
azure-cosmos==3.2.0
azure-cosmosdb-nspkg==2.0.2
azure-cosmosdb-table==1.0.6
azure-functions==1.15.0
azure-graphrbac==0.61.1
azure-identity==1.13.0
azure-keyvault==4.2.0
azure-keyvault-certificates==4.7.0
azure-keyvault-keys==4.8.0
azure-keyvault-secrets==4.7.0
azure-mgmt-advisor==9.0.0
azure-mgmt-apimanagement==1.0.0
azure-mgmt-applicationinsights==1.0.0
azure-mgmt-appplatform==8.0.0
azure-mgmt-authorization==1.0.0
azure-mgmt-batch==15.0.0
azure-mgmt-cdn==10.0.0
azure-mgmt-cognitiveservices==11.0.0
azure-mgmt-compute==19.0.0
azure-mgmt-containerinstance==7.0.0
azure-mgmt-containerregistry==8.0.0b1
azure-mgmt-containerservice==15.1.0
azure-mgmt-core==1.4.0
azure-mgmt-cosmosdb==6.4.0
azure-mgmt-costmanagement==1.0.0
azure-mgmt-databricks==1.0.0b1
azure-mgmt-datafactory==1.1.0
azure-mgmt-datalake-nspkg==3.0.1
azure-mgmt-datalake-store==0.5.0
azure-mgmt-dns==8.0.0b1
azure-mgmt-eventgrid==8.0.0
azure-mgmt-eventhub==8.0.0
azure-mgmt-frontdoor==1.1.0
azure-mgmt-hdinsight==7.0.0
azure-mgmt-iothub==1.0.0
azure-mgmt-keyvault==8.0.0
azure-mgmt-logic==9.0.0
azure-mgmt-managementgroups==1.0.0b1
azure-mgmt-monitor==2.0.0
azure-mgmt-msi==1.0.0
azure-mgmt-network==17.1.0
azure-mgmt-nspkg==3.0.2
azure-mgmt-policyinsights==1.0.0
azure-mgmt-rdbms==8.1.0
azure-mgmt-recoveryservices==2.4.0
azure-mgmt-redhatopenshift==1.2.0
azure-mgmt-redis==12.0.0
azure-mgmt-resource==16.1.0
azure-mgmt-resourcegraph==7.0.0
azure-mgmt-search==8.0.0
azure-mgmt-security==1.0.0
azure-mgmt-servicebus==8.2.0
azure-mgmt-servicefabric==1.0.0
azure-mgmt-sql==1.0.0
azure-mgmt-storage==17.1.0
azure-mgmt-subscription==1.0.0
azure-mgmt-trafficmanager==0.51.0
azure-mgmt-web==2.0.0
azure-nspkg==3.0.2
azure-storage-blob==12.16.0
azure-storage-common==2.1.0
azure-storage-file==2.1.0
azure-storage-file-share==12.12.0
azure-storage-queue==12.6.0
boto3==1.26.157
botocore==1.29.157
c7n==0.9.28
certifi==2023.5.7
cffi==1.15.1
charset-normalizer==3.1.0
click==8.1.3
cryptography==41.0.2
distlib==0.3.6
docutils==0.18.1
idna==3.4
importlib-metadata==5.2.0
isodate==0.6.1
jmespath==1.0.1
jsonschema==4.17.3
msal==1.22.0
msal-extensions==1.0.0
msrest==0.7.1
msrestazure==0.6.4
netaddr==0.7.20
oauthlib==3.2.2
portalocker==2.7.0
pycparser==2.21
pyjwt==2.7.0
pyrsistent==0.19.3
python-dateutil==2.8.2
pytz==2023.3
pyyaml==6.0
requests==2.31.0
requests-oauthlib==1.3.1
s3transfer==0.6.1
setuptools==68.0.0
six==1.16.0
tabulate==0.9.0
typing-extensions==4.6.3
tzdata==2023.3
tzlocal==5.0.1
urllib3==1.26.16
zipp==3.15.0
```


### Policy

```shell
policies:
  - name: untagged-active-buckets
    resource: oci.bucket
    filters:
      - 'defined_tags.governance.contactemail': absent

  - name: tagged-active-buckets
    resource: oci.bucket
    filters:
      - 'defined_tags.governance.contactemail': present
```


### Relevant log/traceback output

```shell
Traceback (most recent call last):
  File "/usr/local/bin/custodian", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/lib/python3.11/site-packages/c7n/cli.py", line 363, in main
    command(config)
  File "/usr/local/lib/python3.11/site-packages/c7n/commands.py", line 128, in _load_policies
    return f(options, list(policies))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/c7n/commands.py", line 337, in report
    do_report(
  File "/usr/local/lib/python3.11/site-packages/c7n/reports/csvout.py", line 98, in report
    rows = formatter.to_csv(records, unique=not options.all_findings)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/c7n/reports/csvout.py", line 220, in to_csv
    uniq = self.uniq_by_id(records)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/c7n/reports/csvout.py", line 202, in uniq_by_id
    rec_id = rec[self._id_field]
             ~~~^^^^^^^^^^^^^^^^
KeyError: 'id'
```


### Extra information or context

_No response_

**Comment by mashariqk** [comment]
Hi @Hotpirsch I was able to reproduce the issue. As you noticed, the issue is happening because there is no id field returned by the api which we are calling for the filtering. We are looking into fixes or workarounds. Thanks

**Comment by Hotpirsch** [comment]
This issue is still not fixed (version 0.1.3) of the provider.

**Comment by vidya-viswanathan** [comment]
@Hotpirsch FYI. The bucket resource that you are trying to filter(as per the policy example mentioned in this issue) doesn't have the support for ID attribute. Each OCI Bucket resource will be identified by the name attribute. Let us know if you need any further clarification.

**Comment by Hotpirsch** [comment]
Okay, maybe I wasn't really clear in the error description. The thing is that I do **not** filter for the ID attribute. I am just running a vanilla report command and it seems like C7N is looking for the "id" key in any case: 
`custodian report -s dashboard-data dashboard-oci-policies.yml --resource oci.bucket --field "State"='lifecycle_state'`
 This command will result in the error reported above. Even if I use the option `--no-default-fields` the error will persist. 
BTW: The standard behaviour with the other providers is that filters for fields that do not exist in the resource.json are simply ignored.

**Comment by kapilt** [comment]
yeah

**Comment by kapilt** [comment]
you'll need to update to 0.1.5 for the fix, 0.1.3 is a few months old, and this issue has been resolved in later releases.
